### PR TITLE
Separate python build workflow into linting and unit testing workflows

### LINKED ISSUE #15: Split python build action into two separate workflows
- [x] <b>Description</b>
    Split python build github into two separate workflows. One for linting checks and one for running unit tests.
- [x] <b>Use cases</b>
It will be easier to find out why the python build is failing.
- [x] <b>Proposed design</b>
    Split python build github into two separate workflows. One for linting checks and one for running unit tests.
- [x] <b>Alternative design approaches</b>
N/a
- [x] <b>Workaround</b>
N/A
- [x] <b>Additional information</b>
N/A

### PR DESCRIPTION
**Pull request by silvs110**
## Description
Closes #15.

Separate python build workflow into linting and unit testing workflows
to provide more visibility of errors.

## Author's Checklist
- [x] **These changes don't break existing dependants.** If they do, and you suggest to proceed, 
please explain why.

## Reviewer's Checklist
- [ ] **All classes have documentation comments.** - NA
- [ ] **All methods have documentation comments.** - NA
- [ ] **No parameters are hardcoded.** Should be loaded as properties instead. If there needs to be hardcoded properties, explain why. - NA
- [ ] **All information to be logged/displayed to the user are internationalized.** If not, explain why. - NA
- [ ] **README.md has been updated if needed.** - NA
- [ ] **setup.py version is set to the latest date** If not, explain why. - NA

## Assignee's Checklist
- [ ] **All GitHub action checks have passed.** - NA
- [x] **All reviewers have approved.**
- [ ] **Relevant documentation has been updated if needed.** - NA
- [ ] **setup.py version is set to the latest date** If not, explain why. - NA

### CODE REVIEW

File: workflow-templates/python_linting.properties.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Python Build - Linting",
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_linting.properties.json:2]:
```suggestion
    "name": "Python linting check - flake8",
```

File: workflow-templates/python_linting.properties.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "description": "pyPadaiyal CI workflow template for linting python builds.",
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_linting.properties.json:3]:
```suggestion
    "description": "pyPadaiyal CI workflow template for checking code lint.",
```

File: workflow-templates/python_linting.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# This workflow will install Python dependencies, run tests and lint with a single version of Python on Mac OS, unbuntu or windows.
+# This workflow will install Python dependencies, run tests and lint with a single version of Python on Mac OS, Ubuntu or Windows.
 
-name: Python Build - clean test verify
+name: Python Build - Linting
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_linting.yml:3]: Change the name accordingly.

File: workflow-templates/python_tests.properties.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Python Build - Test",
+    "description": "pyPadaiyal CI workflow template for testing python builds",
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_tests.properties.json:3]:
```suggestion
    "description": "pyPadaiyal CI workflow template for running the unit tests.",
```

File: workflow-templates/python_tests.properties.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Python Build - Test",
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_tests.properties.json:2]:
```suggestion
    "name": "Python tests - pytest",
```

File: workflow-templates/python_tests.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python on Mac OS, Ubuntu or Windows.
+
+name: Python Build - Test
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_tests.yml:3]: Change the name accordingly.

File: workflow-templates/python_tests.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
```
Comment (the-c0d3br34k3r) [inline workflow-templates/python_tests.yml:13]: Remove the empty line.

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
Reviewed. Requested changes.

File: workflow-templates/python_linting.properties.json (hunk 1 above)
Comment (silvs110) [reply workflow-templates/python_linting.properties.json:2]: done

File: workflow-templates/python_linting.properties.json (hunk 2 above)
Comment (silvs110) [reply workflow-templates/python_linting.properties.json:3]: done

File: workflow-templates/python_linting.yml (hunk 3 above)
Comment (silvs110) [reply workflow-templates/python_linting.yml:3]: done

File: workflow-templates/python_tests.properties.json (hunk 5 above)
Comment (silvs110) [reply workflow-templates/python_tests.properties.json:2]: done

File: workflow-templates/python_tests.properties.json (hunk 4 above)
Comment (silvs110) [reply workflow-templates/python_tests.properties.json:3]: done

File: workflow-templates/python_tests.yml (hunk 6 above)
Comment (silvs110) [reply workflow-templates/python_tests.yml:3]: done

**Review by the-c0d3br34k3r — APPROVED** [review]

**Pull request merged**